app/InterAppCommunication/IACManager.m

```diff
@@ -0,0 +1,260 @@
+//
+//  IACManager.m
+//  IACSample
+//
+//  Created by Antonio Cabezuelo Vivo on 09/02/13.
+//  Copyright (c) 2013 Antonio Cabezuelo Vivo. All rights reserved.
+//
+
+#import "IACManager.h"
+#import "IACDelegate.h"
+#import "IACClient.h"
+#import "IACRequest.h"
+#import "NSString+IACExtensions.h"
+
+
+#if !__has_feature(objc_arc)
+#error InterAppComutication must be built with ARC.
+// You can turn on ARC for only InterAppComutication files by adding -fobjc-arc to the build phase for each of its files.
+#endif
+
+
+NSString * const IACErrorDomain       = @"com.iac.manager.error";
+NSString * const IACClientErrorDomain = @"com.iac.client.error";
+
+// x-callback-url strings
+static NSString * const kXCUPrefix        = @"x-";
+static NSString * const kXCUHost          = @"x-callback-url";
+static NSString * const kXCUSource        = @"x-source";
+static NSString * const kXCUSuccess       = @"x-success";
+static NSString * const kXCUError         = @"x-error";
+static NSString * const kXCUCancel        = @"x-cancel";
+static NSString * const kXCUErrorCode     = @"error-Code";
+static NSString * const kXCUErrorMessage  = @"errorMessage";
+
+// IAC strings
+static NSString * const kIACPrefix       = @"IAC";
+static NSString * const kIACResponse     = @"IACRequestResponse";
+static NSString * const kIACRequest      = @"IACRequestID";
+static NSString * const kIACResponseType = @"IACResponseType";
+static NSString * const kIACErrorDomain  = @"errorDomain";
+
+typedef NS_ENUM(NSUInteger, IACResponseType) {
+    IACResponseTypeSuccess,
+    IACResponseTypeFailure,
+    IACResponseTypeCancel
+};
+
+
+@implementation IACManager {
+    NSMutableDictionary *sessions;
+    NSMutableDictionary *actions;
+}
+
++ (IACManager*)sharedManager {
+    static IACManager *sharedManager = nil;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        sharedManager = [[self alloc] init];
+    });
+    
+    return sharedManager;
+}
+
+- (instancetype)init {
+    self = [super init];
+    if (self) {
+        sessions = [NSMutableDictionary dictionary];
+        actions = [NSMutableDictionary dictionary];
+    }
+    return self;
+}
+
+- (BOOL)handleOpenURL:(NSURL*)url {
+    // An app can respond to multiple url schemes and the app can use different IACManagers for each one
+    // so we test if the url is handled by this manager
+    if (![url.scheme isEqualToString:self.callbackURLScheme]) {
+        return NO;
+    }
+    
+    // If the url is an x-callback-url compatible url we handle it
+    if ([url.host isEqualToString:kXCUHost]) {
+        NSString     *action     = [[url path] substringFromIndex:1];
```

Comment:
There's a couple of places where you alternate between dot syntax and method syntax, such as here. It would be nice if you were consistent–personally, I lean towards dot syntax for most things that are "member-like" as this is (in my mind, this is usually amortized O(1) operations, or those that behave as such).

Reply:
I'm not the "InterAppCommunication" developer. It's an existing Cocoapods library by Antonio Cabezuelo Vivo